modules/xmpp/SdpTransformUtil.js

```diff
@@ -0,0 +1,419 @@
+     * @return {Array.<number>} an array with all SSRC as numbers.
+     */
+    getSSRCs () {
+        return this._ssrcs
+            .map(ssrcInfo => ssrcInfo.id)
+            .filter((ssrc, index, array) => array.indexOf(ssrc) === index);
+    }
+
+    /**
+     * Obtains primary video SSRCs.
+     * @return {Array.<number>} an array of all primary video SSRCs as numbers.
+     * @throws Error if the wrapped media description is not a video.
+     */
+    getPrimaryVideoSSRCs () {
+        const mediaType = this.mLine.type;
+
+        if (mediaType !== 'video') {
+            throw new Error(
+                `getPrimaryVideoSSRCs doesn't work with ${mediaType}`);
+        }
+
+        const videoSSRCs = this.getSSRCs();
+
+        this.forEachSSRCGroup(ssrcGroupInfo => {
+            // Right now, FID groups are the only ones we parse to
+            // disqualify streams.  If/when others arise we'll
+            // need to add support for them here
+            if (ssrcGroupInfo.semantics === "FID") {
+                // secondary FID streams should be filtered out
+                const secondarySsrc = parseSecondarySSRC(ssrcGroupInfo);
+                videoSSRCs.splice(
+                    videoSSRCs.indexOf(secondarySsrc), 1);
+            }
+        });
+        return videoSSRCs;
+    }
+
+    /**
+     * Dumps all SSRC groups of this media description to JSON.
+     */
+    dumpSSRCGroups() {
+        return JSON.stringify(this.mLine.ssrcGroups);
+    }
+
+    /**
+     * Removes all SSRC groups which contain given SSRC number at any position.
+     * @param {number} ssrc the SSRC for which all matching groups are to be
+     * removed.
+     */
+    removeGroupsWithSSRC(ssrc) {
+        if (!this.mLine.ssrcGroups) {
+            return;
+        }
+
+        this.mLine.ssrcGroups = this.mLine.ssrcGroups
+            .filter(groupInfo => groupInfo.ssrcs.indexOf(ssrc + "") === -1);
+    }
+
+    /**
+     * Removes groups that match given semantics.
+     * @param {string} semantics e.g. "SIM" or "FID"
+     */
+    removeGroupsBySemantics(semantics) {
+        if (!this.mLine.ssrcGroups) {
+            return;
+        }
+
+        this.mLine.ssrcGroups
+            = this.mLine.ssrcGroups
+                  .filter(groupInfo => groupInfo.semantics !== semantics);
+    }
+
+    /**
+     * Replaces SSRC (does not affect SSRC groups, but only attributes).
+     * @param {number} oldSSRC the old SSRC number
+     * @param {number} newSSRC the new SSRC number
+     */
+    replaceSSRC(oldSSRC, newSSRC) {
+        if (this.mLine.ssrcs) {
+            this.mLine.ssrcs.forEach(ssrcInfo => {
+                if (ssrcInfo.id === oldSSRC) {
+                    ssrcInfo.id = newSSRC;
+                }
+            });
+        }
+    }
+
+    /**
+     * Executes given call back for each SSRC group
+     * @param {function(object)} callback
+     */
+    forEachSSRCGroup(callback) {
```

Comment:
I want to add getter to expose `ssrcGroups` to the wrapper, but is it ok to add getter without adding a setter at the same time ?